crates/llm-ls/src/adaptors.rs

```diff
@@ -0,0 +1,142 @@
+                APIResponse::Generation(gen) => vec![gen],
+                APIResponse::Generations(gens) => gens,
+                APIResponse::Error(err) => return Err(internal_error(err)),
+            };
+        Ok(generations)
+    }
+}
+
+struct AdaptOllamaRequest;
+impl AdaptOllamaRequest {
+    fn adapt_body(&self, prompt: String, params: CompletionParams) -> Value {
+        let request_body = params.request_body.unwrap_or_default();
+        let body = serde_json::json!({
+            "prompt": prompt,
+            "model": request_body.get("model"),
+        });
+        body
+    }
+    fn adapt_headers(&self) -> Result<HeaderMap, jsonrpc::Error> {
+        let headers = HeaderMap::new();
+        Ok(headers)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct OllamaGeneration {
+    response: String,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+pub enum OllamaAPIResponse {
+    Generation(OllamaGeneration),
+    Error(APIError),
+}
+
+struct AdaptOllamaResponse;
+impl AdaptOllamaResponse {
+    fn adapt_blob(
+        &self,
+        text: Result<String, reqwest::Error>,
+    ) -> Result<Vec<Generation>, jsonrpc::Error> {
+        match text {
+            Ok(text) => {
+                let mut gen: Vec<Generation> = Vec::new();
+                for row in text.split("\n") {
+                    if row.is_empty() {
+                        continue;
+                    }
+                    let chunk = match serde_json::from_str(row) {
+                        Ok(OllamaAPIResponse::Generation(ollama_gen)) => ollama_gen.response,
+                        Ok(OllamaAPIResponse::Error(err)) => return Err(internal_error(err)),
+                        Err(err) => return Err(internal_error(err)),
+                    };
+                    gen.push(Generation {
+                        generated_text: chunk,
+                    })
+                }
+                Ok(gen)
+            }
+            Err(err) => Err(internal_error(err)),
+        }
+    }
+}
+
+const HUGGING_FACE_ADAPTOR: &str = "huggingface";
+
+pub struct Adaptors;
+impl Adaptors {
+    pub fn adapt_body(&self, prompt: String, params: CompletionParams) -> Value {
+        let adaptor = params.adaptor.clone();
+        match adaptor.unwrap_or(HUGGING_FACE_ADAPTOR.to_string()).as_str() {
+            "ollama" => AdaptOllamaRequest.adapt_body(prompt, params),
+            _ => AdaptHuggingFaceRequest.adapt_body(prompt, params),
+        }
+    }
+    pub fn adapt_headers(
+        &self,
+        adaptor: Option<String>,
+        api_token: Option<&String>,
+        ide: Ide,
+    ) -> Result<HeaderMap, jsonrpc::Error> {
+        match adaptor.unwrap_or(HUGGING_FACE_ADAPTOR.to_string()).as_str() {
+            "ollama" => AdaptOllamaRequest.adapt_headers(),
+            _ => AdaptHuggingFaceRequest.adapt_headers(api_token, ide),
+        }
+    }
+    pub fn adapt_blob(
```

Comment:
```suggestion
    pub fn adapt_text(
```
as it's not a blob but the textual response